include/sys/uio.h

```diff
@@ -17,11 +17,20 @@ ssize_t writev(int fd, const struct iovec *iov, int iovcnt);
 
 typedef struct vm_map vm_map_t;
 
+/* This structure stores just enough information about an uio
+ * to be able to restore it after the uio is modified by uiomove(). */
+typedef struct {
+  size_t us_resid;   /* Saved value of uio_resid */
+  int us_iovcnt;     /* Saved value of uio_iovcnt */
+  size_t us_iov_off; /* Saved value of uio_iov_off */
+} uiosave_t;
+
 typedef enum { UIO_READ, UIO_WRITE } uio_op_t;
 
 typedef struct uio {
   iovec_t *uio_iov;      /* scatter/gather list */
   int uio_iovcnt;        /* length of scatter/gather list */
+  size_t uio_iov_off;    /* offset in current iov segment */
```

Comment:
To follow local coding style I'd rename `uio_iov_off` to `uio_iovoff` (look at `uio_iovcnt`). Same for `us_iov_off`.

Reply:
Done